manager/controllers/step_controller.go

```diff
@@ -130,6 +130,7 @@ func (r *StepReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 	step.Status.Selector = selector.String()
 
 	ownerReferences := []metav1.OwnerReference{*metav1.NewControllerRef(step.GetObjectMeta(), dfv1.StepGroupVersionKind)}
+	headlessSvcName := "step-" + pipelineName + "-" + stepName
```

Comment:
Have to add a prefix `step-` to the service name, some of the pipelines (like the ones under ./examples) have their names starting with number (`101-`), which is not allowed in the `Service` name. Adding a prefix to avoid that.